apps/vscode/editor/src/utils/rpc.ts

```diff
@@ -36,7 +36,7 @@ export function rpc(
 		vscode.postMessage(inMessage)
 
 		const handler = ({ data: response }: MessageEvent<ResponseType | ErrorType>) => {
-			if (uuid === response.uuid) {
```

Comment:
this fixes a bug that was preventing bookmark data from loading properly.

the problem was

- our custom rpc infra is set up to give each message a UUID
- after sending a message, it waits for a response
- it was ignoring messages with the same UUID, which makes sense
- but we were sending the response back with the same UUID as the request so it was being ignored
- and so each RPC promise was only being resolved after a _second_ request of the same type was submitted with a different UUID.
- i.e.
    1. await rpc('blah') // sends request A
    2. await rpc('blah') // sends request B, but the result resolves request A
    3. await rpc('blah') // sends request C, but the result resolves request B

and so on.

Reply:
pls add this comment in the code - otherwise, we'll forget why :)